amazon-s3/5.8/modules/ROOT/pages/amazon-s3-connector-reference.adoc

```diff
@@ -0,0 +1,6206 @@
+* S3:PRECONDITION_FAILED
+* S3:BUCKET_ALREADY_EXISTS
+* S3:MISSING_SECURITY_ELEMENT
+* S3:AMBIGUOUS_GRANT_BY_EMAIL_ADDRESS
+* S3:INVALID_URI
+* S3:NOT_SIGNED_UP
+* S3:INVALID_ADDRESSING_HEADER
+* S3:MAX_POST_PRE_DATA_LENGTH_EXCEEDED_ERROR
+* S3:ILLEGAL_VERSIONING_CONFIGURATION_EXCEPTION
+* S3:INVALID_PAYER
+* S3:INVALID_SECURITY
+* S3:NO_SUCH_BUCKET_POLICY
+* S3:OPERATION_ABORTED
+* S3:INVALID_ACCESS_KEY_ID
+* S3:INVALID_LOCATION_CONSTRAINT
+* S3:INLINE_DATA_TOO_LARGE
+* S3:SIGNATURE_DOES_NOT_MATCH
+* S3:UNRESOLVABLE_GRANT_BY_EMAIL_ADDRESS
+* S3:NO_SUCH_LIFECYCLE_CONFIGURATION
+* S3:KEY_TOO_LONG
+* S3:INVALID_BUCKET_NAME
+* S3:INVALID_TOKEN
+* S3:TEMPORARY_REDIRECT
+* S3:BUCKET_NOT_EMPTY
+* S3:MALFORMED_POST_REQUEST
+* S3:BAD_DIGEST
+* S3:CROSS_LOCATION_LOGGING_PROHIBITED
+* S3:REQUEST_TIMEOUT
+* S3:TOO_MANY_BUCKETS
+* S3:NO_SUCH_KEY
+* S3:MISSING_ATTACHMENT
+* S3:INVALID_BUCKET_STATE
+* S3:RETRY_EXHAUSTED
+* S3:INVALID_PART_ORDER
+* S3:INVALID_RESPONSE_PRESIGNEDURL_MALFORMED
+* S3:INVALID_POLICY_DOCUMENT
+* S3:MALFORMED_XML
+* S3:ACCOUNT_PROBLEM
+* S3:USER_KEY_MUST_BE_SPECIFIED
+* S3:MISSING_CONTENT_LENGTH
+* S3:INVALID_DIGEST
+* S3:ACCESS_DENIED
+* S3:MISSING_REQUEST_BODY_ERROR
+* S3:REQUEST_TORRENT_OF_BUCKET_ERROR
+* S3:TOKEN_REFRESH_REQUIRED
+* S3:RESTORE_ALREADY_IN_PROGRESS
+* S3:NO_LOGGING_STATUS_FOR_KEY
+* S3:INVALID_OBJECT_STATE
+* S3:CONNECTIVITY
+* S3:INVALID_ENCRYPTION_ALGORITHM_ERROR
+* S3:PERMANENT_REDIRECT
+* S3:INVALID_REQUEST
+* S3:INCORRECT_NUMBER_OF_FILES_IN_POST_REQUEST
+* S3:UNKNOWN
+* S3:MALFORMED_ACL_ERROR
+* S3:CREDENTIALS_NOT_SUPPORTED
+* S3:NOT_IMPLEMENTED
+* S3:INVALID_TARGET_BUCKET_FOR_LOGGING
+* S3:INVALID_ARGUMENT
+* S3:NO_SUCH_UPLOAD
+* S3:INTERNAL_ERROR
+* S3:SLOW_DOWN
+* S3:METHOD_NOT_ALLOWED
+* S3:ENTITY_TOO_LARGE
+* S3:EXPIRED_TOKEN
+* S3:MISSING_SECURITY_HEADER
+* S3:BUCKET_ALREADY_OWNED_BY_YOU
+* S3:INCOMPLETE_BODY
+
+== Sources
+
+[IMPORTANT]
+You must perform additional configuration to use the sources provided by Amazon S3 Connector. Refer to the xref:amazon-s3-connector-config-topics.adoc#prereq[prerequisites] before configuring your sources.
+
+* <<deleted-object-trigger>>
+* <<new-object-trigger>>
+
+
+
+[[deleted-object-trigger]]
+=== On Deleted Object
+
+`<s3:deleted-object-trigger>`
+
+[IMPORTANT]
+You must perform additional configuration to use the sources provided by Amazon S3 Connector. Refer to the xref:amazon-s3-connector-config-topics.adoc#prereq[prerequisites] before configuring your sources.
+
+Initiates a Mule flow when an object is deleted from an Amazon S3 bucket, and sends an SQS notification about the deleted object.
+
+You must also add the following permissions for this source:
+
+[cols="1,1"]
+|===
+|Specified *Notification Queue Name*
+a|
+* `sqs: ReceiveMessage`
+
+* `sqs: DeleteMessage`
+
+|Specified *Notification Configuration Name*
+a|
+* `s3:GetBucketNotification`
+
+|Unspecified *Notification Queue Name*
+a|
+* `sqs:CreateQueue`
+
+* `sqs: GetQueueAttributes`
```

Comment:
I notice that some of these like `sqs:CreateQueue` do not have a space after the colon and some like `sqs: GetQueueAttributes` do. I'm assuming that's intentional.